appyx-interactions/common/src/commonMain/kotlin/com/bumble/appyx/interactions/core/ui/state/BaseMutableUiState.kt

```diff
@@ -39,12 +45,23 @@ abstract class BaseMutableUiState<TargetUiState>(
         )
     )
 
-    abstract val modifier: Modifier
+    private val sizeChangedModifier: Modifier = Modifier
+        .onSizeChanged { size ->
+            _size.update { size }
+        }
+
+    protected abstract val combinedMotionPropertyModifier: Modifier
+
+    val modifier
+        get() = combinedMotionPropertyModifier
+            .then(sizeChangedModifier)
 
     private val _isBoundsVisible = MutableStateFlow(false)
     private val visibilitySources: Iterable<StateFlow<Boolean>> =
         motionProperties.mapNotNull { it.isVisibleFlow } + _isBoundsVisible
 
+    protected val _size = MutableStateFlow(IntSize.Zero)
+
```

Comment:
Ordering them like this reads somewhat better, in the order of being referenced:

```suggestion
    val modifier
        get() = combinedMotionPropertyModifier
            .then(sizeChangedModifier)

    protected abstract val combinedMotionPropertyModifier: Modifier
                
    private val sizeChangedModifier: Modifier = Modifier
        .onSizeChanged { size ->
            _size.update { size }
        }
        
    protected val _size = MutableStateFlow(IntSize.Zero)
    private val _isBoundsVisible = MutableStateFlow(false)
    private val visibilitySources: Iterable<StateFlow<Boolean>> =
        motionProperties.mapNotNull { it.isVisibleFlow } + _isBoundsVisible
```